sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -343,6 +351,15 @@ class SessionCatalog(
     synchronized { currentDb = dbName }
   }
 
+  def setCurrentDatabaseWithoutCheck(db: String): Unit = {
```

Comment:
Is this necessary？

Reply:
Yes. This is because `setCurrentDatabaseW` has a db exists check, and we have already checked before setting the current db state.

Reply:
Maybe we shall remove `requireDbExists` check completely in SessionCatalog but only rely on the checks in the underlying catalog

Reply:
Probably we cannot as this will be a API behavior change.....

Reply:
It's a different topic as the issue here is we should check database existence with the actual implementation of `spark_catalog`, not `v1SessionCatalog`.

Reply:
Shall we make it `private[sql]` to avoid exposing it too early to users via `spark.sessionState.catalog.setCurrentDatabaseWithoutCheck`?

Reply:
this is already in a private package `catalyst` but I don't have a strong opinion on this.